pypit/artrace.py

```diff
@@ -573,20 +514,24 @@ def trace_slits(slf, mstrace, det, pcadesc="", maskBadRows=False):
         # which means the code skips user-definition for the first detector
         # but adds one for the 2nd.
         ledge, redge = (det-1)*2, (det-1)*2+1
-        if slf._argflag['trace']['orders']['sng_slit'][redge] > 0:
+        if settings.argflag['trace']['slits']['single'][redge] > 0:
             msgs.warn("Using input slit edges on detector {:d}: [{:g},{:g}]".format(
                     det,
-                    slf._argflag['trace']['orders']['sng_slit'][ledge],
-                    slf._argflag['trace']['orders']['sng_slit'][redge]))
+                    settings.argflag['trace']['slits']['single'][ledge],
+                    settings.argflag['trace']['slits']['single'][redge]))
             msgs.warn("Better know what you are doing!")
-            edgearr[:, slf._argflag['trace']['orders']['sng_slit'][ledge]] = -1
-            edgearr[:, slf._argflag['trace']['orders']['sng_slit'][redge]] = +1
+            edgearr[:, settings.argflag['trace']['slits']['single'][ledge]] = -1
+            edgearr[:, settings.argflag['trace']['slits']['single'][redge]] = +1
     else:
         # Even better would be to fit the filt/sqrt(abs(binarr)) array with a Gaussian near the maximum in each column
         msgs.info("Detecting slit edges")
         sqmstrace = np.sqrt(np.abs(binarr))
         for ii in range(medrep):
-            sqmstrace = ndimage.median_filter(sqmstrace, size=(3, 5))
+            sqmstrace = ndimage.median_filter(sqmstrace, size=(3, 7))
+        # Make sure there are no spuriously low pixels
+        sqmstrace[np.where((sqmstrace < 1.0) & (sqmstrace >= 0.0))] = 1.0
```

Comment:
The np.where are superfluous, i.e. this works:

sqmstrace[(sqmstrace < 1.) & (sqmstrace >= 0.)] = 1.